frontend/javascripts/oxalis/controller/combinations/skeletontracing_plane_controller.js

```diff
@@ -340,11 +345,51 @@ function onClick(
   }
 }
 
-function setWaypoint(position: Vector3, ctrlPressed: boolean): void {
+function onRightClick(
+  planeView: PlaneView,
+  position: Point2,
+  shiftPressed: boolean,
+  altPressed: boolean,
+  ctrlPressed: boolean,
+  plane: OrthoView,
+  isTouch: boolean,
+  event: MouseEvent,
+  showNodeContextMenuAt: (number, number, ?number, Vector3, Vector3) => void,
+) {
+  const state = Store.getState();
+  if (isMagRestrictionViolated(state)) {
+    // The current zoom value violates the specified magnification-restriction in the
+    // task type. Therefore, we abort the action here.
+    // Actually, one would need to handle more skeleton actions (e.g., deleting a node),
+    // but not all (e.g., deleting a tree from the tree tab should be allowed). Therefore,
+    // this solution is a bit of a shortcut. However, it should cover 90% of the use case
+    // for restricting the rendered magnification.
+    // See https://github.com/scalableminds/webknossos/pull/4891 for context and
+    // https://github.com/scalableminds/webknossos/issues/4838 for the follow-up issue.
+    return;
+  }
   const { activeViewport } = Store.getState().viewModeData.plane;
   if (activeViewport === OrthoViews.TDView) {
     return;
   }
+
+  const { volume } = state.tracing;
+  if (!volume || volume.activeTool === VolumeToolEnum.MOVE) {
+    // We avoid creating nodes when in brushing mode.
+    const nodeId = event.shiftKey
+      ? maybeGetNodeIdFromPosition(planeView, position, plane, isTouch)
+      : null;
+    const rotation = getRotationOrtho(activeViewport);
```

Comment:
What do you think about passing `activeViewport` to `showNodeContextMenuAt` so that this function can calculate the rotation on its own. This feels more general to me (in case the active viewport is later needed for something else), as we don't pass derived data but instead the "source data".

Reply:
I do not have a strong opinion about this. My goal here was that the context menu component simply gets all the needed values. 

The way you suggested seems fine to me too :+1: